### PR TITLE
first stab to recover dump_dynamical_matrices in public branch

### PR DESCRIPTION
**Pull request by mjv500**
grabbed handy tool from old develop branches. Moved to src, and incorporated in build_things

### CODE REVIEW

**Review by flokno — CHANGES_REQUESTED** [review]
As I understand, the code currently reads in the force constants and dumps to an Abinit DDB file. Would `convert_forceconstant_to_abinit_ddb` be a better name, in line with https://github.com/ollehellman/tdep-devel/tree/master/src/convert_abinit_ddb_to_forceconstant ? 

Since this is a feature exclusive for Abinit, I'm thinking if it would be better to move this to a `utilities` folder that is optionally compiled.

Some comments are given in line

File: tests/dump_dynamical_matrices/test_dump_dynamical_matrices.py
[the remark this answers is not in the document]
Comment (mjv500) [reply tests/dump_dynamical_matrices/test_dump_dynamical_matrices.py:1]: what are you running beyond "make"?

File: src/dump_dynamical_matrices/ddb_io_out.F90
[the remark this answers is not in the document]
Comment (mjv500) [reply src/dump_dynamical_matrices/ddb_io_out.F90:1]: removed

File: tests/dump_dynamical_matrices/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+testfiles:
+	dump_dynamical_matrices
```
[the remark this answers is not in the document]
Comment (mjv500) [reply tests/dump_dynamical_matrices/Makefile]: done

**Comment by mjv500** [comment]
> As I understand, the code currently reads in the force constants and dumps to an Abinit DDB file. Would `convert_forceconstant_to_abinit_ddb` be a better name, in line with https://github.com/ollehellman/tdep-devel/tree/master/src/convert_abinit_ddb_to_forceconstant ?

name can stay unchanged - the idea is others beyond abinit may want their own export  options, and the infrastructure is there.

File: tests/dump_dynamical_matrices/test_dump_dynamical_matrices.py
[the remark this answers is not in the document]
Comment (flokno) [reply tests/dump_dynamical_matrices/test_dump_dynamical_matrices.py:1]: https://github.com/tdep-developers/tdep/tree/main/tests#run-tests

**Review by flokno — APPROVED** [review]
looks good

**Comment by flokno** [comment]
Merge @mjv500 ?

**Comment by mjv500** [comment]
Thanks for the time and nagging flokno!

**Pull request merged**